f3/api.go

```diff
@@ -46,9 +46,45 @@ type Clock interface {
 
 type Signer interface {
 	// Signs a message for the given sender ID.
-	Sign(sender ActorID, msg []byte) []byte
+	Sign(sender ActorID, msg []byte) ([]byte, error)
 	// Verifies a signature for the given sender ID.
-	Verify(sender ActorID, msg, sig []byte) bool
+	Verify(sender ActorID, msg, sig []byte) error
+}
+
+// AggSignatureScheme encapsulates signature aggregation functionality.
+// The implementation is expected to be aware of a mapping between ActorIDs and public keys.
```

Comment:
As above, I don't think we want this implicit awareness. The API should be purely in terms of keys.